projects/kubernetes-sigs/image-builder/build/setup_packer_configs.sh

```diff
@@ -160,14 +160,33 @@ if [ "$IMAGE_FORMAT" = "ova" ] && \
         if [ -z $PACKER_ACTIVE_INTERFACE ]; then
             echo "Finding interface for Packer temporary HTTP server"
             if [ "$(uname -s)" = "Linux" ]; then
-                if command -v route &> /dev/null; then
-                    PACKER_ACTIVE_INTERFACE=$(route |awk '/^default/{print $NF}')
+                # NOTE: not all Ubuntu variant comes "route" command by default, better to detect with "ip".
+                if command -v ip &> /dev/null; then
```

Comment:
any reason not to just do both, if route exists use it otherwise check for ip?

Reply:
+1

Reply:
It's simply because we had `ip` elsewhere in the script. I can do it for both.... should I ?

Or maybe we should rely on info from /sys/class/... so that we could get rid of binary deps, WDYT?

Reply:
Let me check if there is a way to get default route without binary requirements. I will get you back later.

Reply:
How about this?

### Checking with `ip`

```
$ ip route show default
default via 192.168.96.1 dev eth0
```

### Checking with `route`
```
$ route -n
Kernel IP routing table
Destination     Gateway         Genmask         Flags Metric Ref    Use Iface
0.0.0.0         192.168.96.1    0.0.0.0         UG    0      0        0 eth0
169.254.169.254 0.0.0.0         255.255.255.255 UH    0      0        0 eth0
169.254.170.23  0.0.0.0         255.255.255.255 UH    0      0        0 pod-id-link0
192.168.96.0    0.0.0.0         255.255.224.0   U     0      0        0 eth0
192.168.119.120 0.0.0.0         255.255.255.255 UH    0      0        0 eni7e18413c9aa
192.168.122.116 0.0.0.0         255.255.255.255 UH    0      0        0 enia4662b17240
```

### Checking with `/proc/net/route`
```
$ cat /proc/net/route
Iface	Destination	Gateway 	Flags	RefCnt	Use	Metric	Mask		MTU	Window	IRTT
eth0	00000000	0160A8C0	0003	0	0	0	00000000	0	0	0
eth0	FEA9FEA9	00000000	0005	0	0	0	FFFFFFFF	0	0	0
pod-id-link0	17AAFEA9	00000000	0005	0	0	0	FFFFFFFF	0	0	0
eth0	0060A8C0	00000000	0001	0	0	0	00E0FFFF	0	0	0
eni7e18413c9aa	7877A8C0	00000000	0005	0	0	0	FFFFFFFF	0	0	0
enia4662b17240	747AA8C0	00000000	0005	0	0	0	FFFFFFFF	0	0	0
```

### Extracting default route info from `/proc/net/route`
```
$ cat /proc/net/route | awk '$2 == "00000000" { print $1 }'
eth0
```

Or even more stricter
```
$ cat /proc/net/route | awk '$2 == "00000000" { print $1 }' | head -1
eth0
```

Reply:
Test with `Ubuntu 22.04`

```
ubuntu@ip-172-31-42-28:~$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 22.04.3 LTS
Release:	22.04
Codename:	jammy
```

```
ubuntu@ip-172-31-42-28:~$ ip route show default
default via 172.31.32.1 dev ens5 proto dhcp src 172.31.42.28 metric 100
```

```
ubuntu@ip-172-31-42-28:~$ route -n
Command 'route' not found, but can be installed with:
sudo apt install net-tools
```

```
ubuntu@ip-172-31-42-28:~$  cat /proc/net/route
Iface	Destination	Gateway 	Flags	RefCnt	Use	Metric	Mask		MTU	Window	IRTT
ens5	00000000	01201FAC	0003	0	0	100	00000000	0	0	0
ens5	02001FAC	01201FAC	0007	0	0	100	FFFFFFFF	0	0	0
ens5	00201FAC	00000000	0001	0	0	100	00F0FFFF	0	0	0
ens5	01201FAC	00000000	0005	0	0	100	FFFFFFFF	0	0	0
```

```
ubuntu@ip-172-31-42-28:~$ cat /proc/net/route | awk '$2 == "00000000" { print $1 }'
ens5
```

```
ubuntu@ip-172-31-42-28:~$ cat /proc/net/route | awk '$2 == "00000000" && $8 == "00000000" { print $1 }' | head -1
ens5
```

Reply:
This seems reasonable, thanks for testing on multiple platforms!  Technically we support building images on al2/ubuntu 20 + 22 and rhel, any chance you can try on rhel real quick? if not im sure its fine.

I would change the code slightly to do your new check and then if its empty for any reason fallback to the known interface names, instead of like we do now checking if `ip` is available and only falling back if its not.

Reply:
verified on an rhel node that your code works there too!  lets move forward with that approach.

Reply:
Thanks for helping on verifying with RHEL, will submit an update for this later.